### ISSUE TITLE
Bug combining min_epochs and EarlyStopping

### ISSUE THREAD

**Issue by Laolga** [Lightning-AI/lightning#18308]
### Bug description

When Trainer is getting the stop signal due to no further improvements, but min. epochs are not reached it starts doing arbitrary number of interactions (see transition from epoch 9 to epoch 10): 

```
Epoch 6: 100%|██████████| 142/142 [00:27<00:00,  5.21it/s, loss=1.24, v_num=0, val_f1=0.594]
Metric val_f1 improved by 0.006 >= min_delta = 0.0. New best score: 0.594
Epoch 9: 100%|██████████| 142/142 [00:27<00:00,  5.23it/s, loss=1.2, v_num=0, val_f1=0.589] 
Monitored metric val_f1 did not improve in the last 3 records. Best score: 0.594. Signaling Trainer to stop.
Epoch 10:   0%|          | 0/142 [00:00<?, ?it/s, loss=1.2, v_num=0, val_f1=0.589]         
Trainer was signaled to stop but the required `min_epochs=15` or `min_steps=None` has not been met. Training will continue...
Epoch 10: : 3852it [06:13, 10.30it/s, loss=1.2, v_num=0, val_f1=0.578]                       
Monitored metric val_f1 did not improve in the last 4 records. Best score: 0.594. Signaling Trainer to stop.
```

All next epochs also run for 3852 iterations without progress bar

### What version are you seeing the problem on?

v1.9

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

cc @carmocca @awaelchli

**Comment by awaelchli** [comment]
Hi @Laolga 
It would be useful to know how to reproduce it. Otherwise, it won't be possible for us to work on this. This might have already been fixed by #16719 but I'm not sure if it's the same issue.

**Comment by awaelchli** [comment]
Fixed in https://github.com/Lightning-AI/lightning/pull/16719. Upgrading to 1.9.2 or later should make it work for you.
